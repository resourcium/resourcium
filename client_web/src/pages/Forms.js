import React from 'react';

import { WELLBEING_LINKS } from 'src/common/config';

import './Forms.scss';
//https://forms.office.com/Pages/ShareFormPage.aspx?id=jnDz8R-0wUKyZ6gJDYDz7Ee6R-0qnPpDsQbYeCN2iARUMzRRN002U1dGUUU0OU5GMTNPNko1VFRHQi4u&sharetoken=qznLlgcV42CHf3KnyR2y
//This is the pain points form duplication link

//https://forms.office.com/Pages/ShareFormPage.aspx?id=jnDz8R-0wUKyZ6gJDYDz7Ee6R-0qnPpDsQbYeCN2iARUOEdGUU1WTUQxVlRMTkpJNVo5Mk02TVhGMC4u&sharetoken=Oj3A2AbH6qqOrUXnZ4Wh
//This is the form for additional help duplication link.

export default function FormPage() {
    return (
        <div className="FormsPage page">
            <h2>Student Wellbeing page</h2>
            <div className="intro">
                <p>
                    Use this page to report your stress levels, course blockers
                    as well as areas where you require help in. Below are the
                    forms that are currently available to you.
                </p>
            </div>
            {WELLBEING_LINKS.map((link, i) => (
                <React.Fragment key={i}>
                    <hr />
                    <div className="FormSection">
                        <p>{link.description}</p>
                        <a href={link.href} target="_blank" rel="noreferrer">{link.label}</a>
                    </div>
                </React.Fragment>
            ))}
        </div>
    );
}
