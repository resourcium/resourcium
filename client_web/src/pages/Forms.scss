.FormsPage {
    a {
        display: block;
        height: 70px;
        padding: 0;
        line-height: 70px;
        width: 100%;

        border-radius: 5px;
        font-weight: 500;

        background-color: var(--button-bgd-color);
        color: var(--button-text-color);
        font-size: 25px;
        text-align: center;
    }

    .intro {
        background-color: var(--forms-bgd-color);
        color:var(--forms-txt-color);
        padding: 5px;
    }

    .FormSection {
        background-color: var(--forms-bgd-color);
        color: var(--forms-txt-color);
        padding: 5px;
    }

    p {
        margin: 10px;
    }
}
