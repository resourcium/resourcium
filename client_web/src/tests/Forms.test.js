import { create } from 'react-test-renderer';
import React from 'react';

import Forms from 'src/pages/Forms';

import { login } from 'src/tests/util';

test('forms snapshot matches', async () => {
    expect(create(await login(<Forms />))).toMatchSnapshot();
});
