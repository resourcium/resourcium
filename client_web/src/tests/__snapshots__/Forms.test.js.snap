// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forms snapshot matches 1`] = `
<div
  className="FormsPage page"
>
  <h2>
    Student Wellbeing page
  </h2>
  <div
    className="intro"
  >
    <p>
      Use this page to report your stress levels, course blockers as well as areas where you require help in. Below are the forms that are currently available to you.
    </p>
  </div>
  <hr />
  <div
    className="FormSection"
  >
    <p>
      Use this form as a means of getting help for additional course content and material that you need. Before you use this form, head over to the Student Help Page for resources that the university already provides.
    </p>
    <a
      href="https://forms.office.com/Pages/ResponsePage.aspx?id=_oivH5ipW0yTySEKEdmlwn_uYgKALZRKrImf9y-xIrtURTRWMk9HTzFHVTNVUU5BQ1pIRDBaTTZPWi4u"
      rel="noreferrer"
      target="_blank"
    >
      Additional Help Form
    </a>
  </div>
  <hr />
  <div
    className="FormSection"
  >
    <p>
      Use this form to emphasise your stress levels at the moment on the course your doing. This will allow course reps to gain insights on how students are doing as a whole anonymously and bring about improvements.
    </p>
    <a
      href="https://forms.office.com/Pages/ResponsePage.aspx?id=_oivH5ipW0yTySEKEdmlwn_uYgKALZRKrImf9y-xIrtUNFg2REQ5MjVFMjZTRVpIS1BYWVczNk5LQy4u"
      rel="noreferrer"
      target="_blank"
    >
      Report Stress Level
    </a>
  </div>
</div>
`;
